### PR TITLE
Publish unit test results

### PR DESCRIPTION
**Pull request by asuresh4**
**Description:** Publish unit test results to CircleCI. Prep for #1029, with the idea of being able to retrieve test metadata useful for report generation.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
@jchengsfx please review.

File: .circleci/config.yml [hunk 1]
```diff
@@ -193,7 +193,12 @@ jobs:
       - attach_to_workspace
       - run:
           name: Unit tests
-          command: make test
+          command: |
+               mkdir -p unit-test-results/junit
+               trap "go-junit-report  -set-exit-code < unit-test-results/go-unit-tests.out > unit-test-results/junit/results.xml" EXIT
+               make test | tee unit-test-results/go-unit-tests.out
+      - store_test_results:
```
Comment (jchengsfx) [inline .circleci/config.yml:200]: Should we also save the test reports with `store_artifacts`?

**Comment by jrcamp** [comment]
lgtm, agree it wouldn't hurt to store in artifacts.

**Comment by asuresh4** [comment]
Looks like CI is hanging on this despite the rebase, will try opening a new PR.

**Pull request closed without merging**